libuuid fails to build on linux

This is the output:

```
[20:54:18] [DEBU] Running command with direct output: cd '/home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/source/libuuid' && make clean
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/source/libuuid/missing aclocal 
 cd . && /bin/bash /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/source/libuuid/missing automake --foreign
configure.ac:8: error: required file './compile' not found
configure.ac:8:   'automake --add-missing' can install 'compile'
make: *** [Makefile:395: Makefile.in] Error 1
[20:54:19] [ERRO] Uncaught SPC\exception\RuntimeException: Command run failed with code[2]: cd '/home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/source/libuuid' && make clean at /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/globals/functions.php(100)
[20:54:19] [ERRO] #0 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/util/UnixShell.php(47): f_passthru()
#1 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/builder/unix/library/libuuid.php(33): SPC\util\UnixShell->exec()
#2 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/builder/LibraryBase.php(144): SPC\builder\linux\library\libuuid->build()
#3 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/builder/LibraryBase.php(152): SPC\builder\LibraryBase->tryBuild()
#4 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/builder/unix/UnixBuilderBase.php(143): SPC\builder\LibraryBase->tryBuild()
#5 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/command/BuildCliCommand.php(143): SPC\builder\unix\UnixBuilderBase->buildLibs()
#6 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/src/SPC/command/BaseCommand.php(99): SPC\command\BuildCliCommand->handle()
#7 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/vendor/symfony/console/Command/Command.php(326): SPC\command\BaseCommand->execute()
#8 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/vendor/symfony/console/Application.php(1078): Symfony\Component\Console\Command\Command->run()
#9 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#10 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#11 /home/runner/work/durable-php/durable-php/cli/dist/static-php-cli/bin/spc(21): Symfony\Component\Console\Application->run()
```

--- a/config/lib.json
+++ b/config/lib.json
@@ -353,6 +353,9 @@
         "source": "libuuid",
         "static-libs-unix": [
             "libuuid.a"
+        ],
+        "headers": [
+            "uuid/uuid.h"
         ]
     },
     "libuv": {

--- a/config/source.json
+++ b/config/source.json
@@ -385,7 +385,7 @@
     },
     "libuuid": {
         "type": "git",
-        "url": "https://github.com/cloudbase/libuuid.git",
+        "url": "https://github.com/static-php/libuuid.git",
         "rev": "master",
         "license": {
             "type": "file",

--- a/src/SPC/builder/unix/library/libuuid.php
+++ b/src/SPC/builder/unix/library/libuuid.php
@@ -10,29 +10,37 @@ use SPC\store\FileSystem;
 
 trait libuuid
 {
-    public function patchBeforeBuild(): bool
-    {
-        FileSystem::replaceFileStr($this->source_dir . '/configure', '-${am__api_version}', '');
-        return true;
-    }
-
     /**
      * @throws FileSystemException
      * @throws RuntimeException
      */
     protected function build(): void
     {
-        shell()->cd($this->source_dir)
-            ->exec('chmod +x configure')
-            ->exec('chmod +x install-sh')
+        FileSystem::resetDir($this->source_dir . '/build');
+        shell()->cd($this->source_dir . '/build')
             ->exec(
-                './configure ' .
-                '--enable-static --disable-shared ' .
-                '--prefix='
+                'cmake ' .
+                "{$this->builder->makeCmakeArgs()} " .
+                '..'
             )
-            ->exec('make clean')
-            ->exec("make -j{$this->builder->concurrency}")
-            ->exec('make install DESTDIR=' . BUILD_ROOT_PATH);
-        $this->patchPkgconfPrefix(['uuid.pc']);
+            ->exec("cmake --build . -j {$this->builder->concurrency}");
+        copy($this->source_dir . '/build/libuuid.a', BUILD_LIB_PATH . '/libuuid.a');
+        FileSystem::createDir(BUILD_INCLUDE_PATH . '/uuid');
+        copy($this->source_dir . '/uuid.h', BUILD_INCLUDE_PATH . '/uuid/uuid.h');
+        $pc = FileSystem::readFile($this->source_dir . '/uuid.pc.in');
+        $pc = str_replace([
+            '@prefix@',
+            '@exec_prefix@',
+            '@libdir@',
+            '@includedir@',
+            '@LIBUUID_VERSION@',
+        ], [
+            BUILD_ROOT_PATH,
+            '${prefix}',
+            '${prefix}/lib',
+            '${prefix}/include',
+            '1.0.3',
+        ], $pc);
+        FileSystem::writeFile(BUILD_LIB_PATH . '/pkgconfig/uuid.pc', $pc);
     }
 }

--- a/src/globals/test-extensions.php
+++ b/src/globals/test-extensions.php
@@ -13,14 +13,14 @@ declare(strict_types=1);
 
 // If you want to test your added extensions and libs, add below (comma separated, example `bcmath,openssl`).
 $extensions = match (PHP_OS_FAMILY) {
-    'Linux', 'Darwin' => 'dba',
-    'Windows' => 'mbstring,pdo_sqlite,mbregex,dba',
+    'Linux', 'Darwin' => 'uuid',
+    'Windows' => 'mbstring,pdo_sqlite,mbregex',
 };
 
 // If you want to test lib-suggests feature with extension, add them below (comma separated, example `libwebp,libavif`).
 $with_libs = match (PHP_OS_FAMILY) {
-    'Linux', 'Darwin' => 'qdbm',
-    'Windows' => 'qdbm',
+    'Linux', 'Darwin' => '',
+    'Windows' => '',
 };
 
 // Please change your test base combination. We recommend testing with `common`.

--- a/src/globals/tests/uuid.php
+++ b/src/globals/tests/uuid.php
@@ -1,0 +1,6 @@
+<?php
+
+declare(strict_types=1);
+
+assert(function_exists('uuid_create'));
+assert(strlen(uuid_create(0)) === 36);

